cap-primitives/src/rustix/fs/stat_unchecked.rs

```diff
@@ -15,5 +23,26 @@ pub(crate) fn stat_unchecked(
         FollowSymlinks::No => AtFlags::SYMLINK_NOFOLLOW,
     };
 
+    // `statx` is preferred on Linux because it can return creation times.
+    // Linux kernels prior to 4.11 don't have `statx` and return `ENOSYS`. We
+    // store the availability in a global to avoid unnecessary syscalls.
+    #[cfg(all(target_os = "linux", target_env = "gnu"))]
+    static HAS_STATX: AtomicBool = AtomicBool::new(true);
+
+    #[cfg(all(target_os = "linux", target_env = "gnu"))]
+    if HAS_STATX.load(Ordering::Relaxed) {
+        let statx_result = statx(
+            start,
+            path,
+            atflags,
+            StatxFlags::BASIC_STATS | StatxFlags::BTIME,
+        );
+        match statx_result {
+            Ok(statx) => return Ok(MetadataExt::from_rustix_statx(statx)),
+            Err(rustix::io::Error::NOSYS) => HAS_STATX.store(false, Ordering::Relaxed),
```

Comment:
I saw a comment in `open_impl.rs` about seccomp on Android making ENOSYS a bad idea. I don't think that applies here yet, but it might later. Is there a better approach? Should I place a comment on this landmine?

Reply:
Rust's own standard library calls `statx` without doing an explicit version check, so we're ok on Androidd here.

Looking into it though, I notice that there is a potential problem with `statx` returning `EPERM` on old versions of Docker; std does a [special check](https://github.com/rust-lang/rust/pull/65685/) to avoid this problem. If you want to implement that same trick here, that'd be great, though since rustix [also needs to do this](https://github.com/bytecodealliance/rustix/issues/352), it's also fine to leave this code as-is for now, and I'll handle this when I update rustix.

Reply:
Nice catch. It seems like if we merge this PR now, it'd break metadata for not-very old versions of Docker, so that's probably bad. 

Why/where does rustix need this logic? I thought it just wrapped `statx` and doesn't really care whether or not `statx` succeeds.

Reply:
Rustix uses `statx` in its implementation of `rustix::fs::stat`. Rustix usually doesn't do that kind of emulation, but in this case, it's motivated by a desire to be y2038-ready on platforms which support that. On Linux, that means using `statx` instead of the native `stat` family of syscalls whenever possible. So if you want to fix this now, that'd be great, but I am also happy to fix this myself at the same time that I fix rustix.

Reply:
Ah, I didn't see that earlier because I was searching through an older version, oops.

Reply:
I just simulated the seccomp statx -> EPERM behavior on my system using this Python script (as root):
```python
import errno
import os
import seccomp

f = seccomp.SyscallFilter(seccomp.ALLOW)
f.add_rule(seccomp.ERRNO(errno.EPERM), "statx")
f.load()

os.execl(
    "target/debug/deps/fs_additional-d7e3e72f4c824747",
    "fs_additional",
    "--nocapture",
    "--test-threads=1",
    "metadata_vs_std_fs",
)
```

It's not quite as accurate as getting an old Docker version, but I'd rather not do that :)

Reply:
I updated `stat_unchecked` to handle `EPERM`. I couldn't do the same `EFAULT` trick as Rust `std` uses (I think it'd require unsafe), but I pushed the test off the critical path. Let me know what you think.

Reply:
I'm considering adding something in Rustix to allow or the null-pointer `EFAULT` trick, but for now the code you have here looks good.